ingester/src/data.rs

```diff
@@ -787,7 +783,23 @@ impl TableData {
             .create_or_get(partition_key, sequencer_id, self.table_id)
             .await
             .context(CatalogSnafu)?;
-        let data = PartitionData::new(partition.id);
+
+        // get info on the persisted parquet files to use later for replay or for snapshot
+        // information on query.
+        let files = repos
+            .parquet_files()
+            .list_by_partition_not_to_delete(partition.id)
+            .await
+            .context(CatalogSnafu)?;
+        // for now we just need the max persisted
+        let max_persisted_sequence_number = files
+            .iter()
+            .max_by(|x, y| x.max_sequence_number.cmp(&y.max_sequence_number))
+            .map(|p| p.max_sequence_number);
```

Comment:
Perhaps we could swap these lines around, extracting the sequence number first and then `max()`ing those.

```suggestion
            .map(|p| p.max_sequence_number)
            .max();
```